### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/DirectSum/Finsupp): add some more variants of `finsuppTensorFinsupp`

### PR DESCRIPTION
**Pull request by acmepjz**
- add `finsuppTensorFinsuppLid`, `finsuppTensorFinsuppRid` as well as their simp lemmas
- make `finsuppTensorFinsupp'` a special case of `finsuppTensorFinsuppLid`
- add `TensorProduct.lid_eq_rid`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Related: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Linearly.20disjoint/near/411691388

does not depend on: #10824 #11635 [probably?]

### CODE REVIEW

File: Mathlib/LinearAlgebra/DirectSum/Finsupp.lean [hunk 1]
```diff
@@ -68,24 +68,86 @@ theorem finsuppTensorFinsupp_symm_single (i : ι × κ) (m : M) (n : N) :
...
+    (finsuppTensorFinsupp' R ι κ).symm (Finsupp.single i (r₁ * r₂)) =
+      Finsupp.single i.1 r₁ ⊗ₜ Finsupp.single i.2 r₂ :=
+  finsuppTensorFinsuppLid_symm_single R R ι κ i r₁ r₂
+
+-- -- cannot add this because we need to import Mathlib.Data.Finsupp.Pointwise
+-- @[simp]
+-- theorem finsuppTensorFinsupp'_symm_single_mul_single (i : ι × κ) (r₁ r₂ : R) :
+--     (finsuppTensorFinsupp' R ι κ).symm (Finsupp.single i r₁ * Finsupp.single i r₂) =
+--       Finsupp.single i.1 r₁ ⊗ₜ Finsupp.single i.2 r₂ := by
+--   rw [← Finsupp.single_mul, finsuppTensorFinsupp'_symm_single]
```
Comment (alreadydone) [inline Mathlib/LinearAlgebra/DirectSum/Finsupp.lean:143]: Does this import create a loop? If not I think it's appropriate to add it.

**Review by alreadydone — APPROVED** [review]

File: Mathlib/LinearAlgebra/DirectSum/Finsupp.lean (hunk 1 above)
Comment (acmepjz) [reply Mathlib/LinearAlgebra/DirectSum/Finsupp.lean:143]: In fact I doubt the usefulness of this result, in particular as a simp lemma: suppose it was `Finsupp.single i r * Finsupp.single i (s * t)`, then probably `simp` will expand the second `Finsupp.single` and skip this result.

Perhaps we should remove `finsuppTensorFinsupp'_symm_single_mul_single` in favor of `finsuppTensorFinsupp'_symm_single` and asks use to manually `rw` to desired form.

Comment (acmepjz) [reply Mathlib/LinearAlgebra/DirectSum/Finsupp.lean]: I removed this, but added two more results which I need to use later. What's your opinion?

**Review by alreadydone — APPROVED** [review]
Thanks, LGTM
maintainer merge

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**